### PR TITLE
kubelet/rkt: merge environments instead of overriding.

### PR DESCRIPTION
**Pull request by yifan-gu**

### CODE REVIEW

**Comment by yifan-gu** [comment]
This PR changes the on-disk systemd service file name. Before this, all pod with the same UID has the same service file name `foo.service`. This causes at least 2 problems when pod foo restarts for some reason: 
- 1. We lose the old pod's service file, thus the info about this dead pod.
- 2. We will fail to run the new service file because of the naming collision. (Not sure why systemd.reload doesn't work here).

The concern is we will leave moreservice files on disk, but this shouldn't be an issue once we are able to gc them(as well as containers, images).
/cc @dchen1107 @yujuhong

**Comment by yifan-gu** [comment]
Related to #8951

**Comment by dchen1107** [comment]
Above lgtm is not from me. The only thing I did is just open this pr. :-(

**Comment by thockin** [comment]
Dawn, you should open a ticket with GitHub to see where that came from.

@brendandburns maybe we should halt bots until we understand this?

**Comment by dchen1107** [comment]
Since I am OOT since tomorrow, reassign it to @yujuhong Thanks!

**Comment by yifan-gu** [comment]
@dchen1107  Travel safe :)

File: pkg/kubelet/rkt/rkt.go [hunk 1]
```diff
@@ -190,11 +191,31 @@ func (r *runtime) runCommand(args ...string) ([]string, error) {
 	return strings.Split(strings.TrimSpace(string(output)), "\n"), nil
 }
 
-// makePodServiceFileName constructs the unit file name for a pod using its UID.
-func makePodServiceFileName(uid types.UID) string {
+// makePodServiceFileName constructs the unit file name for a pod using
+// its pod UID and rkt UUID. We use the rkt UUID here because it's not
+// clear how to update and reload the service file once the service remains
+// after exited.
```
Comment (yujuhong) [inline pkg/kubelet/rkt/rkt.go]: If I understand correctly, you simply want a unique ID for the file name. Why would you need to update the service file?

**Comment by yujuhong** [comment]
LGTM with one question

File: pkg/kubelet/rkt/rkt.go (hunk 1 above)
Comment (yifan-gu) [inline pkg/kubelet/rkt/rkt.go]: @yujuhong Good question. So the situation here is 1 kube pod can map to 2 rkt pods (imagine the first rkt pod crashed, and we are about to start the second one).
For some reason(I don't recall now, but seems to be related to `RemainAfterExit`) it's not updated. Let's just hold one and let me just try `ReloadOrRestartUnit`.

Comment (yujuhong) [inline pkg/kubelet/rkt/rkt.go]: Sure. SGTM.

**Comment by yifan-gu** [comment]
@yujuhong PR updated. Removed some unnecessary functions. Now one pod will only have one service file to map with. If the service file's content is updated, we will invoke 'systemctl daemon-reload' (which equals to dbus.Conn.Reload()) to reload the service file and then restart the service.
Upon deletion, the service file will be removed as well.

**Comment by yujuhong** [comment]
LGTM.

**Pull request merged**